### PR TITLE
Add support for GL_PROGRAM_POINT_SIZE

### PR DESCRIPTION
**Pull request by meh**
For #1026.

### CODE REVIEW

**Comment by meh** [comment]
Need feedback on the presence of the getter/setter methods for `point_size`, and on the naming of the fields and methods.

File: src/program/program.rs [hunk 1]
```diff
@@ -84,16 +86,18 @@ impl Program {
                     shaders_store
                 };
 
-                try!(RawProgram::from_shaders(facade, &shaders_store, has_geometry_shader,
-                                              has_tessellation_shaders, transform_feedback_varyings))
+                (try!(RawProgram::from_shaders(facade, &shaders_store, has_geometry_shader,
+                                               has_tessellation_shaders, transform_feedback_varyings)),
+                uses_point_size)
```
Comment (tomaka) [inline src/program/program.rs]: Ugly alignment here

File: src/program/program.rs [hunk 2]
```diff
@@ -262,6 +267,14 @@ impl Program {
     pub fn get_shader_storage_blocks(&self) -> &HashMap<String, UniformBlock> {
         self.raw.get_shader_storage_blocks()
     }
+
+    pub fn uses_point_size(&self) -> bool {
+      self.uses_point_size
+    }
+
+    pub fn set_point_size_usage(&mut self, value: bool) {
```
Comment (tomaka) [inline src/program/program.rs]: I don't think it should have a setter. That's something that never changes.

**Comment by tomaka** [comment]
Could you:
 1 - Remove the setter.
 2 - Add a comment at `ProgramExt::use_program` that says that it enables/disables GL_POINT_SIZE.
 3 - Add a check in `Program::new` to check that this option is supported by the backend, and a new variant in `ProgramCreationError`.

**Comment by tomaka** [comment]
4 - Add a line in the CHANGELOG

**Comment by meh** [comment]
1 - I thought about the setter for the `Binary` variant, or would there be another way to set it for it? Otherwise should I remove the field from that variant and just default to false?

3 - what should the check be? I keep failing at finding minimum version for stuff, teach me your ways :panda_face:

File: src/program/program.rs (hunk 1 above)
Comment (meh) [inline src/program/program.rs]: How should I align it? Just add the missing space to indent beyond the `(`?

**Comment by tomaka** [comment]
To find the minimum version I just go on docs.gl and open each version one by one.
When I'm not sure (there are some errors on this site) I take a look at the official specs.

Here it's minimum OpenGL 3.0, and OpenGL ES doesn't support it.

**Comment by meh** [comment]
Updated.

**Comment by tomaka** [comment]
Thanks!

**Pull request merged**